### PR TITLE
Explain integration sprint 27

### PR DESCRIPTION
**Pull request by sopel39**

### CODE REVIEW

**Comment by sopel39** [comment]
Added skewness stats to explain analyze.

FYI: @KBP-TDC @ilfrin @pnowojski @cberner

File: presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java [hunk 1]
```diff
@@ -382,17 +390,24 @@ public OperatorStats getOperatorStats()
             info = infoSupplier.get();
         }
 
+        double hashCollisions = (double) this.hashCollisions.getTotalCount();
```
Comment (pnowojski) [inline presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java]: don't hide/overwrite scopes of the variables

File: presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java [hunk 2]
```diff
@@ -83,6 +84,8 @@
     private final AtomicLong finishCpuNanos = new AtomicLong();
     private final AtomicLong finishUserNanos = new AtomicLong();
 
+    private final CounterStat hashCollisions = new CounterStat();
```
Comment (pnowojski) [inline presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java]: Why do you need `CounterStat` for this? Simple `AtomicLong` or double isn't enough?

File: presto-main/src/main/java/com/facebook/presto/sql/planner/PlanPrinter.java [hunk 3]
```diff
@@ -1084,46 +1181,135 @@ public PlanNodeId getPlanNodeId()
...
+                    operatorTotalDrivers,
+                    operatorInputPositions, operatorInputPositionsSquared,
+                    operatorWeightedHashCollisions, operatorWeightedHashCollisionsSquared);
         }
     }
+
+    private static double computerStdDev(double sumSquared, double sum, long n)
+    {
+        double average = sum / n;
+        return sqrt(max((sumSquared - 2 * sum * average + average * average * n) / n, 0d));
```
Comment (pnowojski) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/PlanPrinter.java]: comment why do you need max?

File: presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java [hunk 4]
```diff
@@ -30,6 +30,7 @@
 
 import java.lang.management.ManagementFactory;
 import java.lang.management.ThreadMXBean;
+import java.math.BigInteger;
```
Comment (pnowojski) [inline presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java]: Do you really need BigInteger for this?

Comment (sopel39) [inline presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java]: using double

File: presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java (hunk 2 above)
Comment (sopel39) [inline presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java]: changed to `AtomicLong` . They use CounterStats though for some stats.

File: presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java (hunk 1 above)
Comment (sopel39) [inline presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java]: done

**Pull request closed without merging**